engine/src/main/java/org/camunda/bpm/engine/identity/PasswordPolicyRule.java

```diff
@@ -43,13 +45,38 @@
   Map<String, String> getParameters();
 
   /**
+   * <p>
+   *   <strong>
+   *     DEPRECATED: Use {@link PasswordPolicyRule#execute(String, User)} instead!
```

Comment:
I understand that the design decision was to not mark the method as deprecated via the `@Deprecated` annotation, as we are forced to support it nevertheless. I would challenge that by arguing that it does add the value of discouraging the use of the method while not forcing a user to switch necessarily. I think it would be good to use the annotation and explain in the javadocs, why it is discouraged to use it. 

On another note, mentioning in the javadocs that it is deprecated does not add any additional value compared to using the annotation and might be even more confusing (It says the method is deprecated, but it is not).

Either way, I think we should explain in which cases the it is fine to stick with the old (deprecated) method and when a user should not use it anymore (e.g. when using user data in a PasswordPolicyRule implementation).

We can discuss this if you like.

Reply:
I agree that a Javadocs comment brings no value. However, I would still argue against the `@Deprecated` annotation since public API is never removed from Camunda BPM 7.x, and we guarantee that it continues to work in future minor and patch releases as initially designed. Under these given circumstances, deprecating the API method would just bother our users and brings no extra value.